src/backend/oper.d

```diff
@@ -0,0 +1,401 @@
+        OPpredec,               // --x overloading
+
+        OPva_start,             // va_start intrinsic (dmd)
+
+        OPMAX                   // 1 past last operator
+}
+
+/* Convert from token to assignment operator    */
+//int asgtoktoop(int tok) { return tok + OPeq - TKeq; }
+
+
+/************************************
+ * Determine things about relational operators.
+ */
+
+//OPER rel_toktoop(int tk) { return tk - TKle + RELOPMIN; }
+
+extern __gshared ubyte[RELOPMAX - RELOPMIN + 1]
+        _rel_not,
+        _rel_swap,
+        _rel_integral,
+        _rel_exception,
+        _rel_unord;
+
+int rel_not(int op)       { return _rel_not      [op - RELOPMIN]; }
+int rel_swap(int op)      { return _rel_swap     [op - RELOPMIN]; }
+int rel_integral(int op)  { return _rel_integral [op - RELOPMIN]; }
+int rel_exception(int op) { return _rel_exception[op - RELOPMIN]; }
+int rel_unord(int op)     { return _rel_unord    [op - RELOPMIN]; }
+
+/****************************************
+ * Conversion operators.
+ * Convert from conversion operator to conversion index
+ * parallel array invconvtab[] in cgelem.c)
+ */
+int convidx(OPER op) { return op - CNVOPMIN; }
+
+
+/**********************************
+ * Various types of operators:
+ *      OTbinary        binary
+ *      OTunary         unary
+ *      OTleaf          leaf
+ *      OTcommut        commutative (e1 op e2) == (e2 op e1)
+ *                      (assoc == !=)
+ *      OTassoc         associative (e1 op (e2 op e3)) == ((e1 op e2) op e3)
+ *                      (also commutative)
+ *      OTassign        assignment = op= i++ i-- i=-i str=
+ *      OTpost          post inc or post dec operator
+ *      OTeop0e         if (e op 0) => e
+ *      OTeop00         if (e op 0) => 0
+ *      OTeop1e         if (e op 1) => e
+ *      OTsideff        there are side effects to the operator (assign call
+ *                      post ?: && ||)
+ *      OTconv          type conversion operator that could appear on lhs of
+ *                      assignment operator
+ *      OTlogical       logical operator (result is 0 or 1)
+ *      OTwid           high order bits of operation are irrelevant
+ *      OTopeq          an op= operator
+ *      OTop            an operator that has a corresponding op=
+ *      OTcall          function call
+ *      OTrtol          operators that evaluate right subtree first then left
+ *      OTrel           == != < <= > >= operators
+ *      OTrel2          < <= > >= operators
+ *      OTdef           definition operator (assign call post asm)
+ *      OTae            potential common subexpression operator
+ *      OTboolnop       operation is a nop if boolean result is desired
+ */
+
+extern (C)
+{
+    static if (__VERSION__ > 2066)
+    {
+        extern __gshared ubyte[OPMAX] optab1, optab2, optab3;
+        extern __gshared ubyte[OPMAX] opcost;
+    }
+    else
+    {
+        extern __gshared ubyte[179] optab1, optab2, optab3;
```

Comment:
See https://github.com/dlang/dmd/pull/6034/commits/962ef0614645ff313a4e69120ceb57eaad2ba5e4#diff-c787edcd24b34ddec8409246ca1c5e4bR296 for a slightly nicer workaround.

```d

 // Workaround 2.066.x bug by resolving the OPMAX value before using it as dimension.
 static if (__VERSION__ <= 2066):
     private enum computeEnumValue = OPMAX;
```